diff/diff.go

```diff
@@ -51,14 +51,21 @@ type Comparer interface {
 	Compare(ctx context.Context, lower, upper []mount.Mount, opts ...Opt) (ocispec.Descriptor, error)
 }
 
+// ApplyConfig is used to hold parameters needed for a apply operation
+type ApplyConfig struct {
```

Comment:
What's going to go in this?  Right now this is all a no-op

Reply:
We want to use the options for encrypted container image support patches for passing decryption parameters (keys) to the Applier.

Reply:
Just trying to imagine how flexible this will be

Reply:
This is here is our work-in-progress tree: https://github.com/stefanberger/containerd/commits/image-encryption.v5

This patch appears there under the name "Extend Applier's Apply() method with an optional options parameter". On top of it are two more patches where the 1st one is called 'Prepare Apply call to carry dcparameters using the ApplyOpts' and the 2nd one converts the existing code to use this way of passing the parameters ('Pass the dcparameters via the ApplyOpts now and remove old code').